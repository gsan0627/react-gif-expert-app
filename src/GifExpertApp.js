import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Samurai X"]);

  /*const handlerAdd = (e) => {
    setCategories([...categories, "Caballeros del Zodiaco"]);
  };*/

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category, i) => (
          <GifGrid key={category} category={category}>
            {category}
          </GifGrid>
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
