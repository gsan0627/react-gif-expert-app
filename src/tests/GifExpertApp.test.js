import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Component tests <GifExpectApp />", () => {
  test("testing the component", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });
});
