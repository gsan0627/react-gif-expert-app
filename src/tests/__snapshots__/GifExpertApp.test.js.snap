// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component tests <GifExpectApp /> testing the component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Samurai X"
      key="Samurai X"
    >
      Samurai X
    </GifGrid>
  </ol>
</Fragment>
`;

exports[`Pruebas del componente <GifExpectApp /> probando el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Samurai X"
      key="Samurai X"
    >
      Samurai X
    </GifGrid>
  </ol>
</Fragment>
`;
